CMake/SiloConfig.cmake.in

```diff
@@ -0,0 +1,79 @@
+#-----------------------------------------------------------------------------
+# SILO Config file for compiling against SILO build directory
+#-----------------------------------------------------------------------------
+@PACKAGE_INIT@
+
+set(SILO_PACKAGE_NAME "Silo")
+
+set (${SILO_PACKAGE_NAME}_VALID_COMPONENTS static shared)
+
+#-----------------------------------------------------------------------------
+# User Options
+#-----------------------------------------------------------------------------
+set (SILO_BUILD_SHARED_LIBS    @BUILD_SHARED_LIBS@)
+
+set(SILO_ENABLE_SILEX @SILO_ENABLE_SILEX@)
+set(SILO_ENABLE_BROWSER @SILO_ENABLE_BROWSER@)
+set(SILO_ENABLE_FORTRAN @SILO_ENABLE_FORTRAN@)
+set(SILO_ENABLE_HDF5 @SILO_ENABLE_HDF5@)
+set(SILO_ENABLE_JSON @SILO_ENABLE_JSON@)
+set(SILO_ENABLE_PYTHON_MODULE @SILO_ENABLE_PYTHON_MODULE@)
+set(SILO_ENABLE_NETCDF @SILO_ENABLE_NETCDF@)
+set(SILO_ENABLE_TESTS @SILO_ENABLE_TESTS@)
+set(SILO_ENABLE_TESTS_INSTALL @SILO_ENABLE_TESTS_INSTALL@)
+set(SILO_BUILD_FOR_BSD_LICENSE @SILO_BUILD_FOR_BSD_LICENSE@)
```

Comment:
Currently, I think this is kinda sorta handled as sep. tarballs the user downloads. Is this going to allow us to always have *just* the one tarball but build only the bsd-licensed stuff?

Reply:
BTW, this file, SiloConfig.cmake.in is what is used to create the Silo CMake files that get installed and make it easier for consumers (who use cmake) to import Silo.  Real discussion should be about the similar entries in CMakeLists.txt.

Guess a discussion is in order on how best to handle BSD vs non-BSD.
BSD tarball doesn't have all the same source files as non-BSD?
How would we handle SILO_ENABLE_FPZIP and SILO_ENABLE_HZIP which are only enabled for non-BSD?